### PR TITLE
Dev/Integration Environment Setup

### PR DESCRIPTION
**Pull request by ivanligot**
- new project structure to support multiple environments
- added folders for dev environment
- added integration guide (doc/INTEGRATION.md)
- added scripts to generate service accounts
- updated README

### CODE REVIEW

**Comment by wistefan** [comment]
I cannot see any changes to the github actions. How is the deployment of a new application to the cluster triggered?

File: applications_dev/argocd.yaml [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+  labels:
+    purpose: infra
+spec:
+  destination:
+    namespace: argocd
+    server: https://kubernetes.default.svc
+  project: default
+  source:
+    path: ionos_dev/argocd
+    repoURL: https://github.com/ivanligot/dome-gitops.git
```
[the remark this answers is not in the document]
Comment (ivanligot) [reply applications_dev/argocd.yaml:15]: Sure, it should. I simply forgot to update the urls. I'm going to fix them

File: applications_dev/dome-trust/mongodb-trust.yaml [hunk 2]
```diff
@@ -0,0 +1,20 @@
+apiVersion: argoproj.io/v1alpha1
```
[the remark this answers is not in the document]
Comment (ivanligot) [reply applications_dev/dome-trust/mongodb-trust.yaml:1]: At the moment we've just replicated the demo. I can discuss your suggestion with IONOS

File: doc/INTEGRATION.md [hunk 3]
```diff
@@ -0,0 +1,226 @@
...
+## Step 1: Fork the repository and create a new branch
+
+Team members begin by forking the main GitOps repository and creating a new branch for their specific integration work. This facilitates isolated development and changes.
+
+After forking and subsequently cloning the project, navigate to the project directory to proceed with the rest of the guide.
+
+### Repository structure
+The GitOps repository has the following structure:
+
+```
```
[the remark this answers is not in the document]
Comment (ivanligot) [reply doc/INTEGRATION.md:46]: Actually the approach to managing different environments has been subject of several discussions with team IONOS and ACK, and one of the options was to use different branches. In the end, it was agreed to use a single branch to accelerate the release, but of course one of the cons is the code duplication

File: scripts/GenerateAccount.sh [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+
+# For each file in the template directory, apply replace function
+for file in "${templateFiles[@]}"; do
+    replace_namespace_placeholder_in_file "$file" "$outputDirectory"
+done
+
+echo "Manifest files generated: $outputDirectory"
+
+echo "Applying files..."
+
```
[the remark this answers is not in the document]
Comment (ivanligot) [reply scripts/GenerateAccount.sh:49]: You're right, I need to modify the script to ensure that namespaces are handled by ArgoCD as expected (by putting them under ionos/namespaces)

**Comment by ivanligot** [comment]
I haven't updated the workflow yet because we need to add a secret containing the configurations to access the dev environment, and I don't think I have the permissions to do so

**Comment by wistefan** [comment]
> I haven't updated the workflow yet because we need to add a secret containing the configurations to access the dev environment, and I don't think I have the permissions to do so

I added you to the administrators, you should be allowed now

**Review by wistefan — APPROVED** [review]

**Pull request merged**